rosidl_pycommon/rosidl_pycommon/__init__.py

```diff
@@ -67,7 +92,17 @@ def generate_files(
             'Could not find template: ' + template_filename
 
     latest_target_timestamp = get_newest_modification_time(args['target_dependencies'])
-    generated_files = []
+
+    # Generate visibility control files if any
+    generated_files = \
+        [render_visibility_control_file(
+            args['package_name'],
+            template,
+            args['output_dir']
+        )
+            for template in template_basepath.iterdir()
+            if template.is_file() and 'visibility_control' in template.name
+        ]
```

Comment:
@frneer hmm, I know that talking about brittle code here is difficult, but this is brittle. Too many assumptions. 

Why can this header file be just like any other template? There is enough information in template context to expand it.